apps/core/scenes/results/ResultsListItem.js

```diff
@@ -1,61 +1,195 @@
-  /**
-   * TODO: properly handle possible undefined props in final version of list
-   * (for example: should be entry with undefined price still displayed?)
-   *
-   * NOTE: ugly form of string creation below is used for now because
-   * using ?? inside template literals caused problems with code styling
-   * inside VS Code (this part will be probably completely replaced
-   * in next iteration, so it's temporary solution)
-   */
-  const itemString =
-    (flyFrom ?? 'Unknown destination') +
-    ' / ' +
-    (localDeparture ?? 'Unknown departure') +
-    ' -> ' +
-    (flyTo ?? 'Unknown destination') +
-    ' / ' +
-    (localArrival ?? 'Unknown arrival') +
-    ' | ' +
-    (price ?? 'Unknown price');
-  return <Text style={styles.listItem}>{itemString}</Text>;
-};
+  };
 
-const styles = StyleSheet.create({
-  listItem: {
-    fontSize: 12,
-    padding: 2,
-    margin: 2,
-    backgroundColor: defaultTokens.backgroundBody,
-  },
-});
+  // @TODO Handle cases when we don't have some data from GraphQL/REST server
+  getSanitizedRouteItem = (route: RouteItem) => {
+    const timeSimpleFormat = 'H:mm';
+
+    const localArrival = this.parseDate(route?.arrival?.localTime);
+    const localDeparture = this.parseDate(route?.departure?.localTime);
+    const utcArrival = this.parseDate(route?.arrival?.utcTime);
+    const utcDeparture = this.parseDate(route?.departure?.utcTime);
+    const duration =
+      route &&
+      `${DateFNS.differenceInHours(
+        utcArrival,
+        utcDeparture,
+      )}h ${DateFNS.differenceInMinutes(utcArrival, utcDeparture) % 60}m`;
+
+    return {
+      arrival: (route && route.arrival?.city) ?? '',
+      arrivalTime:
+        (localArrival && DateFNS.format(localArrival, timeSimpleFormat)) ?? '',
+      carrier: { code: (route && route.airline) ?? '', name: '' }, // @TODO: get name of the airline
+      tripDate:
+        (localDeparture && DateFNS.format(localDeparture, dateFormat)) ?? '',
+      departure: (route && route.departure?.city) ?? '',
+      departureTime:
+        (localDeparture && DateFNS.format(localDeparture, timeSimpleFormat)) ??
+        '',
+      duration: duration ?? '',
+      id: (route && route.id) ?? '',
+    };
+  };
+
+  getSectorBorderIndex = () => {
+    const { route, routes } = this.props.data;
+    if (!route) {
+      return null;
+    }
+    return route.findIndex(
+      routeItem => routeItem?.departure?.cityCode === routes?.[1]?.[0],
+    );
+  };
+
+  // @TODO: make ConnectionCard more universal (not only for forth and back routes)
+  getSanitizedRoute = (): Array<TripSectorWithId> => {
+    const { route } = this.props.data;
+    if (route) {
+      return route.map((routeItem: RouteItem) =>
+        this.getSanitizedRouteItem(routeItem),
+      );
+    }
+    return [];
+  };
+
+  getSector = (
+    route: Array<TripSectorWithId>, // rename to sanitizeRoute
+    demandedSector: number,
+    sectorBorderIndex: ?number,
```

Comment:
I don't understand that name.

Also, getSector is returning an array of TripSectorWithId, so maybe it should be renamed getSectors?

Reply:
Well, I tried to draw it:
![img_8763](https://user-images.githubusercontent.com/858321/51175204-eb776e80-18b9-11e9-89bc-2afbe68ec4da.jpg)

`getSector` is fine, but `TripSectorWithId` should be renamed to `TripSegmentWithId` and `TripSector` component should be `TripSegment`.

Reply:
Done, renamed